dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -1475,7 +1475,9 @@ void NO_INLINE Aggregator::mergeDataImpl(
     Table & table_src,
     Arena * arena) const
 {
-    for (auto it = table_src.begin(); it != table_src.end(); ++it)
```

Comment:
On the last cycle iteration, calculation of the `table_src.end()` becomes extremely slow(up to 1s) . for template parameters `[with Method = DB::AggregationMethodOneNumber<short unsigned int, HashMapTable<long unsigned int, HashMapCell<long unsigned int, char*, TrivialHash, HashTableNoState>, TrivialHash, HashTableFixedGrower<16>, Allocator<true> > >; Table = HashMapTable<long unsigned int, HashMapCell<long unsigned int, char*, TrivialHash, HashTableNoState>, TrivialHash, HashTableFixedGrower<16>, Allocator<true> >]`

Reply:
I could provide branch with logs  and they output, because such behavior is really strange for me.